src/ray/common/runtime_env_manager.h

```diff
@@ -0,0 +1,56 @@
+// Copyright 2017 The Ray Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include "ray/common/id.h"
+#include "src/ray/protobuf/common.pb.h"
+
+namespace ray {
+
+/// The class that manages runtime env. The lifetime of runtime env is managed
+/// here. There are two places where runtime env need be managed
+///    1) central storage, like GCS or global KV storage
+///    2) local node, where runtime env is fetched
+/// We only track the job and detached actor for runtime env. In summary,
+/// runtime env will be cleaned up when there is no job or detached actor is
+/// using it. The resouce is tracked in uri level.
```

Comment:
Can you clarify whether this class is used locally on nodes or globally? The comment makes it sounds like it is doing both here.

Reply:
Yes. Since we have two places managing runtime env and only deleting method is different, I made one abstraction.

This is a base class. In GCS and raylet, there are two other subclasses which implement different delete handler. Later LRU will be put here too.

I'm not sure whether it's a good place to put this file.